code/integration-tests/runtime-tests/test/tests/pablo/constantProductDex/constantProductTests.ts

```diff
@@ -1,285 +1,1888 @@
+          resultQuoteAmount,
+          resultFee
+        ]
+      } = await sendAndWaitForSuccess(
+        api,
+        walletTrader1,
+        api.events.pablo.Swapped.is,
+        api.tx.pablo.buy(
+          hardCodedPool1.poolId,
+          hardCodedPool1.baseAssetId,
+          {
+            assetId: hardCodedPool1.baseAssetId,
+            amount: transferAmount
+          },
+          false
+        )
+      );
+
+      expect(resultPoolId).to.be.bignumber.equal(new BN(hardCodedPool1.poolId));
+      expect(resultWho.toString()).to.be.equal(api.createType("AccountId32", walletTrader1.publicKey).toString());
+      expect(resultBaseAsset).to.be.bignumber.equal(new BN(hardCodedPool1.baseAssetId));
+      expect(resultQuoteAsset).to.be.bignumber.equal(new BN(hardCodedPool1.baseAssetId));
+      expect(resultBaseAmount).to.be.bignumber.equal(new BN(transferAmount.toString()));
+      expect(resultQuoteAmount).to.be.bignumber.equal(
+        new BN(priceFirstTrade.toFixed(0)).add(resultFee.lpFee.add(resultFee.protocolFee))
+      );
+    });
+
+    it("#4.20  I can not swap with the minimum amount requested being the same as the inAsset.", async function () {
+      // Arbitrary wait because the last test will fail immediately, and can cause this test to run
+      // into priority too low issues.
+      await waitForBlocks(api);
+      const exc = await sendAndWaitForSuccess(
+        api,
+        walletTrader1,
+        api.events.pablo.Swapped.is,
+        api.tx.pablo.swap(
+          hardCodedPool1.poolId,
+          { assetId: 4, amount: Pica(100) },
+          {
+            assetId: 4,
+            amount: Pica(100)
+          },
+          false
+        )
+      ).catch(exc => exc);
+      expect(exc.toString()).to.contain("pablo.CannotRespectMinimumRequested");
+    });
+
+    it("#4.11 I can not buy or swap in a pool that doesn't exist.", async function () {
+      // Arbitrary wait because the last test will fail immediately, and can cause this test to run
+      // into priority too low issues.
+      await waitForBlocks(api);
+      const err = await sendAndWaitForSuccess(
+        api,
+        walletTrader1,
+        api.events.pablo.Swapped.is,
+        api.tx.pablo.buy(1337, 131, { assetId: 1, amount: 10_000 }, false)
+      ).catch(exc => exc);
+      expect(err.toString()).to.contain("pablo.PoolNotFound");
+      const err2 = await sendAndWaitForSuccess(
+        api,
+        walletTrader1,
+        api.events.pablo.Swapped.is,
+        api.tx.pablo.swap(1337, { assetId: 131, amount: 1_000 }, { assetId: 1, amount: 10_000 }, false)
+      ).catch(exc => exc);
+      expect(err2.toString()).to.contain("pablo.PoolNotFound");
+    });
 
     it(
-      "Given that the pool has liquidity and the users have LPTokens, " +
-        "users can remove liquidity from the pool by using LP Tokens",
+      "#4.12 I can buy or swap with the minimum amount requested greater than the trade would give, " +
```

Comment:
Wording seems a bit off, as we are expecting this tx to fail right?

Reply:
You're correct. Will need to re- check and might need to bring this up for the engineering team. 
Ty sir 🙏.